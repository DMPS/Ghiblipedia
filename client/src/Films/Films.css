.films {
  columns: 3 200px;
  -moz-columns: 3 200px;
  -webkit-columns: 3 200px;
  column-gap: 0;
}

Film {
  break-inside: avoid;
  padding: 5px;
}
