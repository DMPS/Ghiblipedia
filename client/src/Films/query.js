import gql from "graphql-tag";

export default gql`
  {
    films {
      director
      producer
      title
      description
    }
  }
`;
