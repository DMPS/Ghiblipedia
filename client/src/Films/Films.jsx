import React, { Component } from "react";
import { Query } from "react-apollo";
import query from "./query";
import Fallback from "../Fallback/Fallback";
import Film from "../Film/Film";
import Loading from "../Loading/Loading";
import "./Films.css";

class Films extends Component {
  render() {
    return (
      <Query query={query}>
        {({ loading, error, data }) => {
          if (loading) return <Loading />;
          if (error) {
            console.error(error);
            return <Fallback />;
          }
          return (
            <div className="films">
              {data.films.map(({ director, producer, title, description }) => {
                return (
                  <Film
                    key={title}
                    director={director}
                    producer={producer}
                    title={title}
                    description={description}
                  />
                );
              })}
            </div>
          );
        }}
      </Query>
    );
  }
}

export default Films;
