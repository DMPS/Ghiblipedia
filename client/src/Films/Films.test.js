import React from "react";
import Films from "./Films";
import query from "./query";
import renderer from "react-test-renderer";

it("Query has not changed", () => {
  expect(query).toMatchSnapshot();
});
