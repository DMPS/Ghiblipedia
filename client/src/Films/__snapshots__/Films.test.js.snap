// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query has not changed 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "films",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "director",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "producer",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "description",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 87,
    "start": 0,
  },
}
`;
